File: app/src/main/java/com/tebet/mojual/view/historysearch/HistorySearchFragmentProvider.java
package com.tebet.mojual.view.historysearch;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class HistorySearchFragmentProvider {

    @ContributesAndroidInjector(modules = HistorySearchFragmentModule.class)
    abstract HistorySearchFragment provideHistorySearchFragmentFactory();
}

File: app/src/main/java/com/tebet/mojual/view/historysearch/HistorySearchFragmentProvider.kt
package com.tebet.mojual.view.historysearch

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class HistorySearchFragmentProvider {

    @ContributesAndroidInjector(modules = [HistorySearchFragmentModule::class])
    internal abstract fun provideHistorySearchFragmentFactory(): HistorySearchFragment
}
